packages/jest-haste-map/src/index.ts

```diff
@@ -250,11 +252,24 @@ class HasteMap extends EventEmitter {
       watch: !!options.watch,
     };
     this._console = options.console || global.console;
-    if (options.ignorePattern && !(options.ignorePattern instanceof RegExp)) {
-      this._console.warn(
-        'jest-haste-map: the `ignorePattern` options as a function is being ' +
-          'deprecated. Provide a RegExp instead. See https://github.com/facebook/jest/pull/4063.',
-      );
+
+    if (options.ignorePattern) {
+      if (options.ignorePattern instanceof RegExp) {
+        this._options.ignorePattern = new RegExp(
+          options.ignorePattern.source.concat('|' + VCS_DIRECTORIES),
+        );
+      } else {
+        const ignorePattern = options.ignorePattern;
+        this._options.ignorePattern = (filePath: string) =>
+          new RegExp(VCS_DIRECTORIES).test(filePath) || ignorePattern(filePath);
```

Comment:
Probably would be a good idea to instantiate this RegExp outside the function, for better performance.